### PR TITLE
feat: implemented the Contributions by type of contributor graph for the list activity page

### LINKED ISSUE #1601: Feature: "Contributions by type of contributor" line chart prototype
### Type of feature

🍕 Feature

### Current behavior

N/A

### Suggested solution

Create a prototype component of the following line chart, with no real data yet, just dummy data.

### Filter values

**Time range filter**
- last 7 days
- last 30 days
- last 3 months

**Contributors filter**
- All contributors
- Active contributors
- New contributors
- Churned contributors

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Created the Contributions by type of contributor for the list activity page. It can be filtered by date range. Note the original graph has been reworked to be an average line chart and not a raw line chart.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #1601

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/833231/f739e4df-ed0b-4591-819b-d0a8a4c46331

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by nickytonline** [comment]
The graphs in some cases look kind of wonky. Discussing this with @jpmcb and @isabensusan.

File: components/molecules/ContributionsEvolutionByTypeCard/contributions-evolution-by-type-card.tsx [hunk 1]
```diff
@@ -1,63 +1,38 @@
...
-  DropdownMenuItem,
-  DropdownMenuTrigger,
-} from "components/atoms/Dropdown/dropdown";
-import PeopleIcon from "img/icons/people.svg";
-import CalendarIcon from "img/calendar.svg";
-import ChevronDownIcon from "img/chevron-down.svg";
-import SVGIcon from "components/atoms/SVGIcon/svg-icon";
 
-const dataTypes = ["active", "new", "churned"] as const;
+const dataTypes = ["active", "new", "alumni"] as const;
```
Comment (nickytonline) [inline components/molecules/ContributionsEvolutionByTypeCard/contributions-evolution-by-type-card.tsx:6]: Our wording and data now use the term `alumni`.

File: components/molecules/ContributionsEvolutionByTypeCard/contributions-evolution-by-type-card.tsx [hunk 2]
```diff
@@ -1,63 +1,38 @@
...
 }
 
-interface Props {
-  data: ContributionEvolutionByTypeDatum[];
+interface ContributionsEvolutionByTypeProps {
+  data?: ContributionEvolutionByTypeDatum[];
+  isLoading: boolean;
 }
 
-const dateFilters = {
```
Comment (nickytonline) [inline components/molecules/ContributionsEvolutionByTypeCard/contributions-evolution-by-type-card.tsx:44]: These filters are no longer necessary. There is a global range filter and the other filter was not supposed to be included as we're showing all contributor types in the graph all the time.

**Comment by nickytonline** [comment]
Just investigating the page freezing. Lots of data in the pipe apparently.

```
You are on the server
Warning: data for page "/lists/[listId]/activity" (path "/lists/8b371419-6219-4448-9de1-f6293a25f98e/activity") is 2.31 MB which exceeds the threshold of 128 kB, this amount of data can reduce performance.
See more info here: https://nextjs.org/docs/messages/large-page-data
```

**Comment by nickytonline** [comment]
> Just investigating the page freezing. Lots of data in the pipe apparently.
> 
> ```
> You are on the server
> Warning: data for page "/lists/[listId]/activity" (path "/lists/8b371419-6219-4448-9de1-f6293a25f98e/activity") is 2.31 MB which exceeds the threshold of 128 kB, this amount of data can reduce performance.
> See more info here: https://nextjs.org/docs/messages/large-page-data
> ```

This doesn't appear to be related to my PR. I happen to load a list that has a lot of data (server-side pros data for other graphs). I'll create a separate issue for this to maybe load that component client-side only.

**Review by jpmcb — APPROVED** [review]
Very solid!!! 👏🏼

**Comment by nickytonline** [comment]
> > Just investigating the page freezing. Lots of data in the pipe apparently.
> > ```
> > You are on the server
> > Warning: data for page "/lists/[listId]/activity" (path "/lists/8b371419-6219-4448-9de1-f6293a25f98e/activity") is 2.31 MB which exceeds the threshold of 128 kB, this amount of data can reduce performance.
> > See more info here: https://nextjs.org/docs/messages/large-page-data
> > ```
> 
> This doesn't appear to be related to my PR. I happened to load a list that has a lot of data (server-side props data for other graphs). I'll create a separate issue for this to maybe load that component's data client-side only. Looks like it's the Treemap component.

Looks like in one of my test lists I had added dependabot who happens to contribute to a few repos. 😅 I removed them from my test list and all good now.

**Review by brandonroberts — APPROVED** [review]
📈

**Comment by nickytonline** [comment]
> The graphs in some cases look kind of wonky. Discussing this with @jpmcb and @isabensusan.

As we discussed, we'll iterate on this graph as need be, but for now, as agreed shipping this.

**Pull request merged**